src/NLog.Targets.Syslog/SyslogTarget.cs

```diff
@@ -75,35 +66,57 @@ protected override void FlushAsync(AsyncContinuation asyncContinuation)
             Task.WhenAll(tasks)
                 .ContinueWith(t =>
                 {
-                    InternalLogger.Debug("[Syslog] Explicit flush completed");
-                    asyncContinuation(t.Exception?.GetBaseException());
-                })
-                .Wait();
+                    var exception = t.Exception?.GetBaseException();
+                    InternalLogger.Debug(exception, "[Syslog] Explicit flush completed");
+                    asyncContinuation(exception);
+                });
+        }
+
+        /// <summary>Closes the target</summary>
+        protected override void CloseTarget()
+        {
+            DisposeDependencies();
+            base.CloseTarget();
         }
 
         /// <summary>Disposes the instance</summary>
         protected override void Dispose(bool disposing)
         {
-            if (disposing)
-            {
-                Enforcement.PropertyChanged -= Init;
-                MessageCreation.PropertyChanged -= Init;
-                MessageSend.PropertyChanged -= Init;
-                DisposeDependencies();
-            }
+            Enforcement.PropertyChanged -= OnConfigurationChanges;
+            MessageCreation.PropertyChanged -= OnConfigurationChanges;
+            MessageSend.PropertyChanged -= OnConfigurationChanges;
+            DisposeDependencies();
             base.Dispose(disposing);
         }
 
+        private void OnConfigurationChanges(object sender, PropertyChangedEventArgs eventArgs)
+        {
+            if (!IsInitialized)
+                return;
+            DisposeDependencies();
+            Init();
+        }
+
+        private void Init()
+        {
+            messageBuilder = MessageBuilder.FromConfig(MessageCreation, Enforcement);
+            asyncLoggers = Enforcement.MessageProcessors.Select(i => new AsyncLogger(Layout, Enforcement, messageBuilder, MessageSend)).ToArray();
+        }
+
         private void DisposeDependencies()
         {
             try
             {
-                Enforcement.MessageProcessors.ForEach(i => asyncLoggers[i].Dispose());
+                Enforcement.MessageProcessors.ForEach(i => asyncLoggers[i]?.Dispose());
```

Comment:
Think removing `if (asyncLoggers != null)` is a bug, since calling DisposeDependencies twice will lead to nullreference-exception.

Reply:
You are right I didn't pay attention to the indexer

Reply:
Just created https://github.com/luigiberrettini/NLog.Targets.Syslog/issues/244